taskcluster/ac_taskgraph/loader/build_config.py

```diff
@@ -13,7 +17,113 @@
 from ..build_config import get_components
 
 
+logger = logging.getLogger(__name__)
+
+ALL_COMPONENTS = object()
+# We always build components that have changed files or dependencies, but we also need
+# to rebuild some or all components when other files change. This structure
+# maintains the latter mapping.
+FILE_PATTERNS_TO_AFFECTED_COMPONENTS = {
+    # Just a placeholder example
+    #"^.*changelog.md.*$": ["build-browser-awesomebar"],
+    "^build.gradle$": ALL_COMPONENTS,
+}
+
+def get_files_changed_pr(base_repository, pull_request_number):
```

Comment:
Maybe put this and the next utility functions in `ac_taskgraph.files_changes` maybe (I suspect they'll eventually want to live in the [corresponding upstream file](https://hg.mozilla.org/ci/taskgraph/file/tip/src/taskgraph/files_changed.py)).